lib/rules/no-empty-first-line/index.js

```diff
@@ -13,23 +11,33 @@ const messages = ruleMessages(ruleName, {
 	rejected: 'Unexpected empty line',
 });
 
-function rule(actual, _, context) {
+/** @type {import('stylelint').Rule} */
+const rule = (primary, _secondaryOptions, context) => {
 	return (root, result) => {
-		const validOptions = validateOptions(result, ruleName, { actual });
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
 
+		// @ts-expect-error -- TS2339: Property 'inline' does not exist on type 'Source'. Property 'lang' does not exist on type 'Source'.
 		if (!validOptions || root.source.inline || root.source.lang === 'object-literal') {
 			return;
 		}
 
-		const rootString = context.fix ? root.toString() : root.source.input.css;
+		const rootString = context.fix ? root.toString() : (root.source && root.source.input.css) || '';
 
 		if (!rootString.trim()) {
 			return;
 		}
 
 		if (noEmptyFirstLineTest.test(rootString)) {
 			if (context.fix) {
-				root.nodes[0].raws.before = root.first.raws.before.trimStart();
+				if (root.first == null) {
+					throw new Error('The root node must have the first node.');
+				}
+
+				if (root.first.raws.before == null) {
+					throw new Error('The first node must have spaces before.');
+				}
+
+				root.first.raws.before = root.first.raws.before.trimStart();
```

Comment:
[note]

- `nodes[0]` -> `first` for consistency
- `root.first` and `root.first.raws.before` can be `undefined`